src/braket/pennylane_plugin/braket_device.py

```diff
@@ -196,13 +197,32 @@ def _braket_to_pl_result(self, braket_result, circuit):
 
         return np.asarray(results)
 
+    @staticmethod
+    def _tracking_data(task):
+        if task.state == "COMPLETED":
```

Comment:
I see this method being invoked as:
```
self._task = self._run_task(self._circuit)
if self.tracker.active:
    tracking_data = self._tracking_data(self._task)
```

`self._run_task` appears to be creating the task but not waiting for completion. Is that correct ? 
If so, would almost everything end up in the failed task set ?

Reply:
I think that is correct, I have now moved the tracking to after waiting for a result.